dbms/src/Storages/Page/V3/PageDirectory.h

```diff
@@ -224,13 +224,15 @@ class VersionedPageEntries
         UInt64 lowest_seq,
         std::map<PageIdV3Internal, std::pair<PageVersion, Int64>> * normal_entries_to_deref,
         PageEntriesV3 & entries_removed,
-        const PageLock & page_lock);
+        const PageLock & page_lock,
+        bool need_fill_entries_removed);
     bool derefAndClean(
         UInt64 lowest_seq,
         PageIdV3Internal page_id,
         const PageVersion & deref_ver,
         Int64 deref_count,
-        PageEntriesV3 & entries_removed);
+        PageEntriesV3 & entries_removed,
+        bool need_fill_entries_removed);
```

Comment:
Make the interface to use type `PageEntriesV3 *` for `entries_removed`, and send `nullptr` when we don't need to fill the vector.
This is a unified style with how we pass the param `normal_entries_to_deref`
```
    bool cleanOutdatedEntries(
        UInt64 lowest_seq,
        std::map<PageIdV3Internal, std::pair<PageVersion, Int64>> * normal_entries_to_deref,
        PageEntriesV3 * entries_removed,
        const PageLock & page_lock);
    bool derefAndClean(
        UInt64 lowest_seq,
        PageIdV3Internal page_id,
        const PageVersion & deref_ver,
        Int64 deref_count,
        PageEntriesV3 * entries_removed);
```